motoko/vetkd/src/app_frontend_js/src/index.js

```diff
@@ -235,10 +235,22 @@ async function ibe_decrypt(ibe_ciphertext_hex) {
 
 document.getElementById("login").onclick = async (e) => {
   e.preventDefault();
+
+  // According to https://github.com/dfinity/internet-identity?tab=readme-ov-file#local-replica,
+  // for local deployments, the II URL must be different depending on the browser:
+  // Chrome, Firefox: http://<canister_id>.localhost:4943
+  // Safari: http://localhost:4943?canisterId=<canister_id>
+  // 
+  // Safari detection rules are according to: https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent#browser_name_and_version
+  let isSafari = /^(?!.*chrome\/\d+)(?!.*chromium\/\d+).*safari\/\d+/i.test(navigator.userAgent);
```

Comment:
Where is that regex from? Just FYI: it does not match my user-agent on Safari, because that contains the `Safari` (capitalized).

Reply:
I crafted it based on the rules for Safari recommended [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent#browser_name_and_version) (See the comment one line above).

The capitalization of `Safari` shouldn't be a problem, because of the `i`nsensitive flag at the very end. Maybe you didn't add a version number, e.g., `/123` in your test?

Do you know of another/better regex that could be used?

Reply:
No the problem is the capitalization of `Safari`: I.e. your regex containing `safari` does not match `Safari`.

But in general I would use [ua-parser](https://www.npmjs.com/package/ua-parser-js).

Reply:
Discussed offline. We'll keep the simple regex for now. Note for readers: ua-parser is [using](https://github.com/faisalman/ua-parser-js/blob/master/LICENSE.md) the AGPL license now (previously MIT).